docs/examples/rx/05_read_storage/index.js

```diff
@@ -0,0 +1,62 @@
+const { combineLatest } = require('rxjs');
+const { switchMap, map } = require('rxjs/operators');
+
+// Import the API
+const { ApiRx } = require('@polkadot/api');
+
+// Our address for Alice on the dev chain
+const Alice = '5GoKvZWG5ZPYL1WUovuHW3zJBWBP5eT8CbqjdRY4Q6iMaDtZ';
+
+// Separate function for retrieval of validators and their associated balance
+function validatorsWithBalance (api) {
+  return api.query.session.validators()
+    .pipe(
+      switchMap((validators) => {
+        // Retrieve the balances for all validators
+        return combineLatest(
+          ...validators.map((authorityId) =>
+            api.query.balances.freeBalance(authorityId)
+              .pipe(
+                map((balance) => ({
+                  address: authorityId.toString(),
+                  balance: balance.toString()
+                }))
+              )
+          )
+        );
+      })
+    );
+}
+
+async function main () {
+  // Instantiate the API with default provider via Promise when ready
+  const api = await ApiRx.create();
+
+  // Subscribe to basic chain state/storage queries
+  // Use the Storage chain state (runtime) Node Interface.
+  const subscriptionApiRx =
+    api
+      .pipe(
+        switchMap((api) =>
+          combineLatest([
+            api.query.system.accountNonce(Alice),
+            api.query.timestamp.blockPeriod(),
+            validatorsWithBalance(api)
+          ])
+        )
+      ).subscribe(([accountNonce, blockPeriod, validatorBalances]) => {
+        console.log(`Account Nonce of ${accountNonce} for Alice with address ${Alice}`);
+        console.log(`Block Period: ${blockPeriod.toNumber()} seconds`);
+        console.log(`Validator Balances: ${JSON.stringify(validatorBalances, null, 2)}`);
+
+        const isReceived = !!accountNonce && !!blockPeriod && !!validatorBalances;
+
+        // Cleanup and unsubscribe from the subscription if all data received
+        if (isReceived) {
+          subscriptionApiRx.unsubscribe();
```

Comment:
If you use `first()` operator, you don't need to manually unsubscribe.

There's a subtle difference though:
- if the node behaves correctly, then the two versions are identical
- however if the node errors, and one of the value is `undefined/null/false`, the subtle difference is that:
  - your version will not unsubscribe, and will wait for a next value. So potentially multiple console.logs.
  - the `first()` version will unsubscribe after the 1st value. So one console.log, will some values equal to `undefined/null/false`

Reply:
i've added this as a "challenge" in the Readme

Reply:
Maybe another option in this example is to replace `combineLatest` with `forkJoin`, since it only takes action when a response has been received for all requests https://www.learnrxjs.io/operators/combination/forkjoin.html